docs/src/components/ButtonLink.tsx

```diff
@@ -0,0 +1,88 @@
+import React from "react";
+import styled, { css } from "styled-components";
+import { PolymorphicPropsWithoutRef } from "react-polymorphic-types";
+
+const DefaultComponent = "a";
+
+interface OwnProps {
+  type: "primary" | "secondary";
+  size?: "medium" | "large";
+  iconLeft?: React.ReactNode;
+  iconRight?: React.ReactNode;
+  children: React.ReactNode;
+}
+
+type Props<T extends React.ElementType = typeof DefaultComponent> = PolymorphicPropsWithoutRef<
+  OwnProps,
+  T
+>;
+
+const StyledComponent = styled(DefaultComponent)<Props>`
+  ${({ theme, type, size = "medium" }) => css`
+    display: inline-flex;
+    align-items: center;
+    ${size === "medium" &&
+    css`
+      padding: 0.875rem 1.5rem;
+    `}
+    ${size === "large" &&
+    css`
+      padding: 1.25rem 2rem;
+    `}
+    border-radius: 100px; /* value safely greater than button height to achieve pill effect */
+    font-weight: 500;
+
+    > * + * {
+      margin-left: 10px;
+    }
+
+    ${type === "primary" &&
+    css`
+      background: ${theme.orbit.backgroundButtonPrimary};
+      color: ${theme.orbit.paletteWhite};
+      &:hover {
+        background: ${theme.orbit.backgroundButtonPrimaryHover};
+      }
+      &:active {
+        background: ${theme.orbit.backgroundButtonPrimaryActive};
+      }
+      &:focus {
+        background: ${theme.orbit.paletteProductNormal};
+      }
+    `}
+
+    ${type === "secondary" &&
+    css`
+      background: ${theme.orbit.paletteProductLight};
+      color: ${theme.orbit.paletteProductDark};
+      &:hover {
+        background: ${theme.orbit.paletteProductLightHover};
+      }
+      &:active {
+        background: ${theme.orbit.paletteProductLightActive};
+      }
+      &:focus {
+        background: ${theme.orbit.paletteProductLightHover};
```

Comment:
i think it would be better with inline ternary conditions, less lines of code

Reply:
You mean `type === "primary" ? <primary color> : <secondary color>`? I would rather not assume we'll only have two types.

Reply:
I don't really see how extracting the styles to `resolve*` functions will make this nicer, I see it only as moving the logic somewhere else. I like to be able to write the code in a way that can be read from top to bottom, and only extract it if it's complicated, and readability could be improved with variable/function naming, or if the file is too large.